### PR TITLE
I298 unify calling cf

### PR DESCRIPTION
**Pull request by Tereza-Pytelova**
Created common function for calling CF `getFunctionResponseJson`.

### CODE REVIEW

File: scripts/wawCommons.py [hunk 1]
```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
             replacedValuesNumber += 1
     return targetJson, replacedValuesNumber
 
+def getFunctionResponseJson(CFNamespaceUrl, username, password, package, functionName, parameters, data):
```
Comment (ondrejtomas) [inline scripts/wawCommons.py]: I would like to change `CFNamespaceUrl` to something that starts with lower letter... we could probably also consider to split it to endpoint and namespace.

File: scripts/functions_test.py [hunk 2]
```diff
@@ -209,20 +206,18 @@ def main(argv):
...
-        if responseContentType != 'application/json':
-            logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, response.text)
-            continue
+
+        testOutputReturnedJson = getFunctionResponseJson(url + '/' + namespace,
+                                               username,
+                                               password,
+                                               (test['cf_package'] if 'cf_package' in test else package),
+                                               (test['cf_function'] if 'cf_function' in test else function),
+                                               '?blocking=true&result=true',
```
Comment (ondrejtomas) [inline scripts/functions_test.py]: I do not like so much that by 'parameters' we actually think query string even with '?' at the beginning... I would rename it to query string or expect really parameters -> dict... and remove the question mark.

File: scripts/wawCommons.py [hunk 3]
```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
...
+        responseContentType = functionResponse.headers.get('content-type')
+        if responseContentType != 'application/json':
+            logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
+            return None
+
+        return functionResponse.json()
+
+    elif functionResponse.status_code == 202: # try once more
+        # 202 Accepted activation request (should not happen while sending 'blocking=true&result=true')
+        logger.warning("Did not receive response from function '%s' in package '%s', trying once more.", functionName, package)
```
Comment (ondrejtomas) [inline scripts/wawCommons.py:557]: when user did not pass 'blocking=true&result=true' we should not print the warning... if we decide to pass params as map. it should be easy to check this

File: scripts/wawCommons.py [hunk 4]
```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
...
+                                        headers={})
+        if functionResponse.status_code == 200:
+
+            responseContentType = functionResponse.headers.get('content-type')
+            if responseContentType != 'application/json':
+                logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
+                return None
+
+            responseJson = functionResponse.json()
+            if not 'result' in responseJson or not responseJson['result'] or not 'payload' in responseJson['result']:
```
Comment (ondrejtomas) [inline scripts/wawCommons.py]: I'm not sure how often can happen that we get bad response format... but if we want to check all in the IF statement... we should also probably check that responseJson['result'] is dict, just for completeness

File: scripts/wawCommons.py [hunk 5]
```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
...
+                 functionName, package, functionResponse.status_code, json.dumps(functionResponse.json(), ensure_ascii=False).encode('utf8'))
+                return None
+            else:
+                return responseJson['result']['payload']
+
+        elif functionResponse.status_code in [403, 404, 408]:
+            # 403 Forbidden (could be just for specific package or function)
+            # 404 Not Found (action or package could be incorrectly specified for given test)
+            # 408 Request Timeout (could happen e.g. for CF that calls some REST APIs, e.g. Discovery service)
+            # 502 Bad Gateway (when the CF raises exception, e.g. bad params where provided)
```
Comment (ondrejtomas) [inline scripts/wawCommons.py]: Please move this comment to else branch below as this is not in the list of codes that match this branch.

File: scripts/wawCommons.py [hunk 6]
```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
...
+            # => We don't want to continue with testing.
+            logger.critical("Unexpected response status from function '%s' in package '%s' with activation id '%s', status code '%d', response: %s",
+                            functionName, package, activationId, functionResponse.status_code, functionResponse.text)
+            sys.exit(1)
+
+    elif functionResponse.status_code in [403, 404, 408]:
+        # 403 Forbidden (could be just for specific package or function)
+        # 404 Not Found (action or package could be incorrectly specified for given test)
+        # 408 Request Timeout (could happen e.g. for CF that calls some REST APIs, e.g. Discovery service)
+        # 502 Bad Gateway (when the CF raises exception, e.g. bad params where provided)
```
Comment (ondrejtomas) [inline scripts/wawCommons.py]: Please move this comment to else branch below as this is not in the list of codes that match this branch.

**Review by ondrejtomas — CHANGES_REQUESTED** [review]

File: scripts/wawCommons.py (hunk 1 above)
Comment (Tereza-Pytelova) [reply scripts/wawCommons.py]: Splitted.

File: scripts/wawCommons.py (hunk 6 above)
Comment (Tereza-Pytelova) [reply scripts/wawCommons.py]: Moved.

File: scripts/wawCommons.py (hunk 5 above)
Comment (Tereza-Pytelova) [reply scripts/wawCommons.py]: Moved.

File: scripts/functions_test.py (hunk 2 above)
Comment (Tereza-Pytelova) [reply scripts/functions_test.py]: Changed to dict.

File: scripts/wawCommons.py (hunk 3 above)
Comment (Tereza-Pytelova) [reply scripts/wawCommons.py:557]: Now this function always passes 'blocking=true&result=true'

File: scripts/wawCommons.py (hunk 4 above)
Comment (Tereza-Pytelova) [reply scripts/wawCommons.py]: Added.

**Review by ondrejtomas — APPROVED** [review]

**Pull request merged**